tests/unit/test_gitlab_http_methods.py

```diff
@@ -129,6 +129,37 @@ def request_callback(request):
     assert len(responses.calls) == calls_before_success
 
 
+@responses.activate
+def test_http_request_with_retry_on_method_for_transient_network_failures_chunked(gl):
```

Comment:
Instead of duplicating almost the entire test function, any chance we could parametrize the one above to raise one of the two exceptions?

Let me know if you need any help with that - here are some docs
https://docs.pytest.org/en/6.2.x/parametrize.html.
And our own example that's slightly similar perhaps:
https://github.com/python-gitlab/python-gitlab/blob/09a973ee379d82af05a5080decfaec16d2f4eab3/tests/unit/test_cli.py#L70-L83

In your case you only need one parameter, the exception raised.

Reply:
👍 I've parameterized the test